src/libexpr/attr-set.hh

```diff
@@ -15,12 +15,16 @@ struct Value;
 /* Map one attribute name to its value. */
 struct Attr
 {
-    Symbol name;
+    /* the placement of `name` and `pos` in this struct is important.
+       both of them are uint32 wrappers, they are next to each other
+       to make sure that Attr has no padding on 64 bit machines. that
+       way we keep Attr size at two words with not wasted space. */
+    SymbolIdx name;
+    PosIdx pos;
     Value * value;
-    ptr<Pos> pos;
```

Comment:
`ptr<T>` was introduced for pos here. I don't think we currently have a use for a non-nullable regular pointer type, so it's possible to remove it.